consensus/src/chained_bft/consensus_types/block.rs

```diff
@@ -90,17 +92,30 @@ pub struct Block<T> {
 /// ExecutedBlocks are managed in a speculative tree, the committed blocks form a chain. Besides
 /// block data, each executed block also has other derived meta data which could be regenerated from
 /// blocks.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug)]
 pub struct ExecutedBlock<T> {
     /// Block data that cannot be regenerated.
     block: Block<T>,
-    /// The state compute results is calculated for all the pending blocks prior to insertion to
+    /// The processed output needed by executor.
+    output: Arc<ProcessedVMOutput>,
+    /// The state compute result is calculated for all the pending blocks prior to insertion to
     /// the tree (the initial root node might not have it, because it's been already
     /// committed). The execution results are not persisted: they're recalculated again for the
     /// pending blocks upon restart.
     compute_result: Arc<StateComputeResult>,
 }
 
+impl<T> PartialEq for ExecutedBlock<T>
+where
+    T: Serialize + Default + CanonicalSerialize + PartialEq,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.block == other.block
```

Comment:
compare the StateComputeResult too

Reply:
Now I'll stick with @zekun000 's comments. I will followup to see how to fix this.